tests/queries/0_stateless/02920_unary_operators_functions.sql

```diff
@@ -0,0 +1 @@
+SELECT NOT (0) + NOT (0);
```

Comment:
I've came across this because the behavior for some queries changed:

```sql
# upstream
SELECT NOT (3-1) = 1
-- SELECT (NOT (3 - 1)) = 1
0

# 23.11
SELECT NOT (3-1) = 1
-- SELECT NOT ((3 - 1) = 1)
1
```

And MySQL also prints 1 - https://sqlfiddle.com/mysql/online-compiler?id=09bbe665-6e2b-4089-9c78-6ff07f457d50

This particular query is invalid for MySQL, but it looks like `not` is "greedy" there.

Any comments on this?